### DISCUSSION TITLE
Welcome to MyFit Discussions!

### DISCUSSION THREAD

**Discussion by WhyAsh5114** [WhyAsh5114/MyFit#44] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by irgpenera** [comment]
Hey, just wanted to ask, what are the pre-requisites for the app to increase the sets of a particular workout? As recommended by Mike. sets/volume are to be increased if there is little pump/disruption from the previous week. I have seen an increase in volume (load/reps) but nothing on sets.

**Reply by yzee-tech** [reply]
Others may confirm, but my experience has been that the app doesn't increase sets at all. The # of sets is something that the user configures as part of the mesocycle, and then the app will recommend increasing weight or the number of reps per set. I think this generally makes sense under the assumption that a user might not have infinitely flexible time at the gym.

**Reply by WhyAsh5114** [reply]
Yeah, I couldn't really finish the progressive overloading for sets part of the algorithm on time and rushed the release of v2. I didn't really expect so many users this quickly so I thought I'd do it later, but now that there are people already using the app, I don't want to risk breaking existing functionality to add the set increase part now.

But this reminds me, I was thinking of changing the algo for v3, instead of using pump/disruption and soreness to manipulate volume, I was thinking about using set performance as the sole indicator for recovery capability. Like why even soreness as a proxy, when we can directly infer recovery from the performance of the working sets, and much more accurately too. For pump and disruption, that stuff is pretty subjective in my personal experience. 

There are even many smart people not comfortable with using these proxies to alter volumes, hopefully I am not putting words in their mouth but people like Menno Henselmans and Milo Wolf. I am still working on the algorithm at the moment, so nothing is set in stone for v3, but these are just some of my current thoughts.

**Comment by yzee-tech** [comment]
I agree that pump/disruption and soreness are really subjective. I
think the difficulty rating, although subjective, is easier for a user
to rate.

**Reply by WhyAsh5114** [reply]
Yeah, an overall difficulty rating of the workout itself is much easier. Maybe multiple high difficulty ratings could nudge the algorithm to increase volume slower than before, and vice versa.

**Comment by ryanisgreat321** [comment]
This app seems great. I'm a little short on time so I decided to ask my questions here. Do you guys have a discord? And if my subscription on rp hypertrophy ran out on week 5 of a 8 week cycle what's the easiest way to move my exercises here? Also what's the easiest way to access with a iPhone?

**Reply by WhyAsh5114** [reply]
Hey there! Thanks for the kind words :)

1. There isn't a Discord but that's a great idea, might make one soon.
2. If you subscription ran out, I don't think there's an easy way to export data from the RP app, (I'm not sure, if there is please lmk), so you'll have to manually create a new routine with similar exercises.
3. Easiest way to access from an iPhone would be through the web app, using the "Add to home screen" on Safari: https://web.dev/learn/pwa/installation#ios_and_ipados_installation

**Reply by ryanisgreat321** [reply]
Done. So as long if I don't delete my safari history or cookies it should be able to save as much progress as I need it to correct? I'm logged into it using my gmail account right now. What I'm asking is do I need to use Google collab, cloning, etc. or will it save everything on its own as long if I don't delete my history.

**Reply by WhyAsh5114** [reply]
No need to setup anything! Your data is saved to the cloud and linked with your Gmail account. You can even use the same account on multiple devices.

**Reply by ryanisgreat321** [reply]
Awesome. Any big differences between this and rp hypertrophy I should know about? Eventually it will start recommending weight increases to me right?

**Reply by WhyAsh5114** [reply]
Yes, after the first week, reps and load will increase according to your preferences set in the mesocycle and the exercise's parameters (rep range).

Main difference between MyFit and RP Hypertrophy app is the absence of soreness and fatigue feedback, the app uses your performance improvements (which is more objective and easier to track) as the sole proxy of your recovery and growth, adjusting everything based on that. It also has quite a few more dials and knobs to turn to allow you to take full control of the program and algorithms that are used. So have fun tinkering!

**Reply by ryanisgreat321** [reply]
Where can I read about how to use all of them?

**Reply by WhyAsh5114** [reply]
There'll be small info buttons across the app, tap on it to learn more, or most of it is inspired from RP's YouTube videos, can use those as a reference too!

**Reply by WhyAsh5114** [reply]
Yes, MyFit takes a more dynamic approach. 

Since not everyone has the same recovery rates for all muscle groups (in general there might be a trend but we can't be sure for all users), MyFit will use your performance changes and dynamically adapt to them. 

If you aren't recovering well enough (not reach your targeted performance increase set during mesocycle creation, can be overriden manually for individual exercises), it'll reduce the increases every week to accommodate that, if you recover too well (beat your targeted performance increase), the algorithm gets incentive to add sets and increase your targets even more.

**Comment by Nkraj03** [comment]
Hey there! I really appreciate you taking your time to create this extremely useful project. I'm happy that I came across this.

I tried this just today and I just wanted to ask- many exercises are missing? Can I add them manually or does it have to be updated by you?

This is the only question I have because every other thing is created to perfection. 

Thanks in advance

**Reply by WhyAsh5114** [reply]
U can add them manually

**Reply by WhyAsh5114** [reply]
You aren't limited to the pre-defined exercises; they are only supposed to be used as a starting point and for quick split building. You can also select one of the pre-defined exercises and then customize them according to your needs.

<img width="384" height="686" alt="image" src="https://github.com/user-attachments/assets/5af0efcf-0502-4b42-8d42-270c92773756" />

**Comment by rambleonnrambleonn** [comment]
How do I message the owner of this app

**Reply by WhyAsh5114** [reply]
Hey there! That would be me :)
You can ask anything here or create an issue under https://github.com/WhyAsh5114/MyFit/issues.
If you're not familiar with GitHub, feel free to join the Discord and we can discuss over there too: https://discord.com/invite/2g9YPD6PQu
